packages/generate-openapi/src/catalog_api.ts

```diff
@@ -0,0 +1,1486 @@
+const EServiceConsumers = z
+  .object({ results: z.array(EServiceConsumer), totalCount: z.number().int() })
+  .passthrough();
+const AttributeSeed = z
+  .object({ id: z.string().uuid(), explicitAttributeVerification: z.boolean() })
+  .passthrough();
+const AttributesSeed = z
+  .object({
+    certified: z.array(z.array(AttributeSeed)),
+    declared: z.array(z.array(AttributeSeed)),
+    verified: z.array(z.array(AttributeSeed)),
+  })
+  .passthrough();
+const EServiceDescriptorSeed = z
+  .object({
+    description: z.string().min(10).max(250).optional(),
+    audience: z.array(z.string()),
+    voucherLifespan: z.number().int().gte(60).lte(86400),
+    dailyCallsPerConsumer: z.number().int().gte(0),
+    dailyCallsTotal: z.number().int().gte(0),
+    agreementApprovalPolicy: AgreementApprovalPolicy,
+    attributes: AttributesSeed,
+  })
+  .passthrough();
+const UpdateEServiceDescriptorSeed = z
+  .object({
+    description: z.string().min(10).max(250).optional(),
+    audience: z.array(z.string()),
+    voucherLifespan: z.number().int().gte(60).lte(86400),
+    dailyCallsPerConsumer: z.number().int().gte(0),
+    dailyCallsTotal: z.number().int().gte(0),
+    agreementApprovalPolicy: AgreementApprovalPolicy,
+    attributes: AttributesSeed,
+  })
+  .passthrough();
+const EServiceDocumentKind = z.enum(["INTERFACE", "DOCUMENT"]);
+const CreateEServiceDescriptorDocumentSeed = z
+  .object({
+    documentId: z.string().uuid(),
+    kind: EServiceDocumentKind,
+    prettyName: z.string(),
+    filePath: z.string(),
+    fileName: z.string(),
+    contentType: z.string(),
+    checksum: z.string(),
+    serverUrls: z.array(z.string()),
+  })
+  .passthrough();
+const UpdateEServiceDescriptorDocumentSeed = z
+  .object({ prettyName: z.string().min(5).max(60) })
+  .passthrough();
+
+export const schemas = {
+  EServiceDescriptorState,
+  AgreementState,
+  EServiceTechnology,
+  EServiceDoc,
+  AgreementApprovalPolicy,
+  Attribute,
+  Attributes,
+  EServiceDescriptor,
+  EService,
+  EServices,
+  EServiceSeed,
+  ResourceId,
+  ProblemError,
+  Problem,
+  UpdateEServiceSeed,
+  EServiceConsumer,
+  EServiceConsumers,
+  AttributeSeed,
+  AttributesSeed,
+  EServiceDescriptorSeed,
+  UpdateEServiceDescriptorSeed,
+  EServiceDocumentKind,
+  CreateEServiceDescriptorDocumentSeed,
+  UpdateEServiceDescriptorDocumentSeed,
+};
+
+const endpoints = makeApi([
+  {
+    method: "get",
+    path: "/eservices",
+    alias: "getEServices",
+    requestFormat: "json",
+    parameters: [
+      {
+        name: "X-Correlation-Id",
+        type: "Header",
+        schema: z.string(),
+      },
+      {
+        name: "X-Forwarded-For",
+        type: "Header",
+        schema: z.string().optional(),
+      },
+      {
+        name: "name",
+        type: "Query",
+        schema: z.string().optional(),
+      },
+      {
+        name: "eservicesIds",
+        type: "Query",
+        schema: z.array(z.string().brand("Test")).default([]).optional(),
```

Comment:
Here I want to test if I'm able to resolve the issue with the decoder declared inside an array